src/KOKKOS/angle_hybrid_kokkos.cpp

```diff
@@ -0,0 +1,224 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   https://www.lammps.org/, Sandia National Laboratories
+   LAMMPS development team: developers@lammps.org
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#include "angle_hybrid_kokkos.h"
+
+#include "atom_kokkos.h"
+#include "atom_masks.h"
+#include "comm.h"
+#include "error.h"
+#include "force.h"
+#include "kokkos.h"
+#include "memory_kokkos.h"
+#include "neighbor_kokkos.h"
+
+#include <cstring>
+
+using namespace LAMMPS_NS;
+
+#define EXTRA 1000
+
+/* ---------------------------------------------------------------------- */
+
+AngleHybridKokkos::AngleHybridKokkos(LAMMPS *lmp) : AngleHybrid(lmp)
+{
+  kokkosable = 1;
+
+  atomKK = (AtomKokkos *) atom;
+  neighborKK = (NeighborKokkos *) neighbor;
+
+  execution_space = Device;
+
+  datamask_read = EMPTY_MASK;
+  datamask_modify = EMPTY_MASK;
+}
+
+/* ---------------------------------------------------------------------- */
+
+AngleHybridKokkos::~AngleHybridKokkos()
+{
+  deallocate();
+}
+
+/* ---------------------------------------------------------------------- */
+
+void AngleHybridKokkos::compute(int eflag, int vflag)
+{
+  // save ptrs to original anglelist
+
+  int nanglelist_orig = neighbor->nanglelist;
+  neighborKK->k_anglelist.sync_device();
+  auto k_anglelist_orig = neighborKK->k_anglelist;
+  auto d_anglelist_orig = k_anglelist_orig.d_view;
+  auto d_nanglelist = k_nanglelist.d_view;
+  auto h_nanglelist = k_nanglelist.h_view;
+
+  // if this is re-neighbor step, create sub-style anglelists
+  // nanglelist[] = length of each sub-style list
+  // realloc sub-style anglelist if necessary
+  // load sub-style anglelist with 3 values from original anglelist
+
+  if (neighbor->ago == 0) {
+    Kokkos::deep_copy(d_nanglelist,0);
+
+    k_map.sync_device();
+    auto d_map = k_map.d_view;
+
+    Kokkos::parallel_for(nanglelist_orig,LAMMPS_LAMBDA(int i) {
+      const int m = d_map[d_anglelist_orig(i,3)];
+      if (m >= 0) Kokkos::atomic_increment(&d_nanglelist[m]);
+    });
+
+    k_nanglelist.modify_device();
+    k_nanglelist.sync_host();
+
+    maxangle_all = 0;
+    for (int m = 0; m < nstyles; m++)
+      if (h_nanglelist[m] > maxangle_all)
+        maxangle_all = h_nanglelist[m] + EXTRA;
+
+    if (k_anglelist.d_view.extent(1) < maxangle_all)
+      MemKK::realloc_kokkos(k_anglelist, "angle_hybrid:anglelist", nstyles, maxangle_all, 3);
```

Comment:
@megmcca also need a corresponding increase to the size of the view here.

Reply:
got it! tested it and output energies for `angle hybrid harmonic charmm` on non-Kokkos, cpu-Kokkos, and gpu-Kokkos compiles match each other.